### ISSUE TITLE
Server crash if Stack item is null

### ISSUE THREAD

**Issue by ultrasn0w** [ForgeEssentials/ForgeEssentials#2066]
Just experienced this crash on my server, I wonder if this is intended or not.
https://gist.github.com/ultrasn0w/7e3cfd73bb602cf5d01a19495878eacf
Using forgeessentials-1.7.10-1.4.4.1164.

**Comment by spacebuilder2020** [comment]
Can you try updating to the latest version of forge essentials (1176) and see if you still crash

**Comment by spacebuilder2020** [comment]
I have looked at the code and this is intended.  If the stack item is null when obtaining the item, it will crash.  However, you can put it in safemode to disable such crashes at a potential cost of overall stability.

Can you elaborate on what you were doing to have FE checking the permission of a null item?

**Comment by spacebuilder2020** [comment]
@ultrasn0w This is not a bug.  It is intended behavior of forge essentials.  See https://github.com/ForgeEssentials/ForgeEssentials/blob/2f3fe422a7dcf8fb0cdb39a2737270606cc9b4cb/src/main/java/com/forgeessentials/protection/ModuleProtection.java#L389.  I am closing this issue.  

Also, can you please give me more information on how this came up.  While the crash itself is not a bug, the method it occurred may be.

I am closing this issue for now.

**Comment by ultrasn0w** [comment]
The Issue occured in conjunction with The Twilight Forest mod, using the Lamp of Cinders to pass the Thorn blocks. They're destroyed on touch then and sometimes they apparently drop a null item.
http://ftb.gamepedia.com/Twilight_Forest_Progression#Troll

If this is intended by FE I should contact the TF dev then.

**Comment by spacebuilder2020** [comment]
I am not sure why it is intended, you would have to ask @olee, but TF is mentioned in the error message so it is probably something related to TF issues.  The way to prevent the crash is to enable safemode.

**Comment by olee** [comment]
This is because TF has just never cared to fix that bug they produce.
Dropping a null item stack is just invalid and that's why FE by default will throw an error in this case as well (or such bugs would go unnoticed forever).

However because no one at TF seems to care to fix this, I added that safe-mode flag so at least FE users can prevent the crashes.

TL;DR: Blame TF - I reported this to them once already but nothing happened.

**Comment by liach** [comment]
You may send pull requests to their repo, I guess.
https://github.com/Benimatic/twilightforest

**Comment by spacebuilder2020** [comment]
Looks like he is finally fixing it.
https://github.com/Benimatic/twilightforest/commit/ff5f7f5f00e4c599640c2566c7e621eb63bf25b8

**Comment by olee** [comment]
Wow, finally :smile: 
Good thing they fixed it.
